doc: Add information about the preview tag (#19949)
nrfconnect/sdk-nrf: doc/nrf/dev_model_and_contributions/code_base.rst

They follow the ``MAJOR.MINOR.PATCH`` pattern (also further called as ``X.Y.Z``), with possible additional postfixes depending on the release.
All releases are accompanied by their respective :ref:`release_notes`.

+----------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+

> |        Release type        |                              Release criteria                              |                                                 Comments                                                                             |

Reviewer: The table header is misleading as what is in the second column are not release criteria. Please merge the "Release and comments" into a single column named "Description".

Reviewer: Rename the first column into "Version"

Author: Added.

|                                | that the version string is a point in between two releases.                    |                                                                                                                                      |
|                                | Such version does not belong to a release and corresponds to the               |                                                                                                                                      |
|                                | current state of development.                                                  |                                                                                                                                      |
+--------------------------------+--------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+

> | ``MAJOR.MINOR.PATCH-previewN`` | Preview tag, marked through a postfix added to the end of the version.         | Not part of the regular release cycle.                                                                                               |

Reviewer: Let's make a combined description as:

> Preview tag, marked through a postfix added to the end of the version. A preview is recommended for development for customers who are in the initial phase of product development and are in need for new features from the SDK. The ``-previewN`` is verified using automated test suites only. Please refer to the changelog for the full list of executed test suites.

They follow the ``MAJOR.MINOR.PATCH`` pattern (also further called as ``X.Y.Z``), with possible additional postfixes depending on the release.
All releases are accompanied by their respective :ref:`release_notes`.

+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+
|        Version                 |                              Description                                                                                             |
+================================+======================================================================================================================================+

> | ``MAJOR``                      | Introduces a large number of substantial changes across the board                                                                    |

Reviewer 2: ```suggestion
| ``MAJOR``                      | Introduces a large number of substantial changes across the board.                                                                   |
```

|                                | Major releases are always accompanied by a :ref:`migration guide <migration_guides>`.                                                |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+
| ``MINOR``                      | Introduces one or more new functionalities.                                                                                          |
|                                | Version number is increased every time a minor release is cut.                                                                       |
|                                | Minor releases are the default types of an |NCS| release and usually are accompanied by a :ref:`migration guide <migration_guides>`. |
|                                | The new functionalities may break existing APIs.                                                                                     |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+

> | ``PATCH``                      | Introduces bugfixes or minor changes to the existing functionalities.                                                                |

Reviewer 3: ```suggestion
| ``PATCH``                      | Introduces bug fixes or minor changes to the existing functionalities.                                                                |
```
As per [Terms](https://nordicsemi.atlassian.net/wiki/spaces/TECHDOC/pages/120298016/Terms+abbreviations+and+spelling+rules#Terms,abbreviations,andspellingrules-B)

|                                | Patch releases only address functional issues and do not introduce new functionalities.                                              |
|                                | Patch releases do not come with :ref:`migration guide <migration_guides>`.                                                           |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+
| ``MAJOR.MINOR.99``             | The ``99`` added at the end of the version number indicates that the version string is a point in between two releases.              |
|                                | Such version does not belong to a release and corresponds to the                                                                     |

> |                                | current state of development.                                                                                                        |

Reviewer 4: Could be moved to the above line?

|                                | The number can be used between minor, major, or patch releases.                                                                      |
+--------------------------------+--------------------------------------------------------------------------------------------------------------------------------------+

> | ``MAJOR.MINOR.PATCH-previewN`` | Preview tag, marked through a postfix added to the end of the version.                                                               |
> |                                | A preview is recommended for development for users who are in the initial phase of product development                               |

Reviewer 3: ```suggestion
| ``MAJOR.MINOR.PATCH-previewN`` | A preview of one or more new functionalities.                                                               |
|                                | A preview tag is recommended for development for users who are in the initial phase of product development                               |
```
To align this with previous entries...

> |                                | and needs to add new features of the |NCS|.                                                                                          |

Reviewer 3: ```suggestion
|                                | and need to add new features of the |NCS|.                                                                                          |
```

The Git tags are composed as follows::

  vX.Y.Z(-rcN|-previewN)

Tags without a suffix correspond to official releases tested and signed by Nordic Semiconductor engineers.
A release candidate suffix ``-rcN`` is attached if the tag identifies a candidate instead of the actual release.

> In between releases, there might be development tags, called Preview tags.

Reviewer 3: ```suggestion
In between releases, there might be development tags, called preview tags.
```
Any reason for capitalization?

These are identified by a ``-previewN`` suffix included after the |NCS| version, for example ``v2.8.0-preview1``.

> Such versions is recommended for development for users who are in the initial phase of product development and needs to add new features of the |NCS|.

Reviewer 2: ```suggestion
Such versions are recommended for development for users in the initial phase of product development and need to add new features of the |NCS|.
```
